File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set(nums)
        res = 0
        for n in s:
            if n - 1 in s:
                continue
            cur = 1
            curNum = n
            while curNum + 1 in s:
                curNum += 1
                cur += 1
            res = max(res, cur)
        return res

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> s = new HashSet<>();
        for (int n: nums) {
            s.add(n);
        }
        int res = 0;
        for (int n: s) {
            if (s.contains(n - 1)) continue;
            int cur = n;
            int curLen = 0;
            while (s.contains(cur)) {
                cur++;
                curLen++;
            }
            res = Math.max(res, curLen);
        }
        return res;
    }
}